## Ideas:
1. Encode gene regulatory network in the structure of the autoencoders to learn a low-dimension representation of the high-dimension gene expression data, followed by comparison of the reconstitution error, compression power, and visualization of bio-structured autoencoder to conventional fully connected autoencoder and random sparsely connected autoencoders. 
2. Leveraging the multi-faceted information available in a biological sample, such as the tissue of origin, tumor or not tumor, age and gender, etc, a multitask neural network could potentially improve the prediction accuracy of each one of the clinically relevant information in a biological sample by learning from underlying similar structure of relevant tasks. 
3. Biological signal deconvolution in gene expression data: combining the biologically encoded autoencoder and multi-task neural network learner in 1 and 2 to simultaneously maximizing the prediction accuracy of relevant biology signals/tasks while minimizing the accuracy irrelevant biological signals/tasks, thus generating a function to map sample gene expression to sample gene expression with the undesired signal removed. This approach has the potential to remove batch effects as well as remove strong biologically signals that cloud underlying more subtle biological processes.
4. Use such gene expression, DNA methylation and copy number variation data as multiple channel of inputs, and the various sample labels such as the tissue of origin, tumor or not tumor, age and gender as multiple tasks to predict, a multi-channel-multi-task neural network with gene regulatory network and gene module information encoded in its structure could then provide biologically interpretable prediction and insights. 

